### PR TITLE
Add SQLite info to glossary list

### LINKED ISSUE #3427: Add SQLite info to glossary list
### What should be done?

In our glossary page https://docs.ferretdb.io/reference/glossary/, we need to include information on 

- SQLite: Mention that we now offer SQLite backend support.

### PR DESCRIPTION
**Pull request by pvinoda**
1. Added SQLite Info to Glossary List

2. Kept the information similar to that of Postgres

# Description

Closes #3427 .

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [x] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [ ] I marked all done items in this checklist.

### CODE REVIEW

**Comment by AlekSi** [comment]
@pvinoda Please click the CLA button in that comment: https://github.com/FerretDB/FerretDB/pull/3593#issuecomment-1765601950

**Comment by pvinoda** [comment]
@AlekSi I accepted the CLA. I see that a PR job is failing. Should I change any assert statements or any reformatting issues as such ?

File: website/docs/reference/glossary.md [hunk 1]
```diff
@@ -146,4 +146,9 @@ See [Operation modes](../configuration/operation-modes.md) for more details.
 An open source relational database.
 FerretDB uses PostgreSQL as a database engine.
 
+#### SQLite
```
Comment (AlekSi) [inline website/docs/reference/glossary.md:149]:
```suggestion
#### SQLite
```

There is a dangling space

**Comment by AlekSi** [comment]
Should be good now :)

**Review by AlekSi — DISMISSED** [review]

File: website/docs/reference/glossary.md [hunk 2]
```diff
@@ -146,4 +146,9 @@ See [Operation modes](../configuration/operation-modes.md) for more details.
...
```
Comment (AlekSi) [inline website/docs/reference/glossary.md:149]:
```suggestion
### S

#### SQLite
```

@Fashander I guess we need that?

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
Thank you!

**Pull request merged**